pkg/northbound/gnmi/set.go

```diff
@@ -149,12 +149,18 @@ func (s *Server) Set(ctx context.Context, req *gnmi.SetRequest) (*gnmi.SetRespon
 			Version:  version,
 		}
 
-		_, errDevice := mgr.DeviceStore.Get(topodevice.ID(target))
-		if errDevice != nil && status.Convert(errDevice).Code() == codes.NotFound {
+		deviceTopo, errDevice := mgr.DeviceStore.Get(topodevice.ID(target))
+		if (errDevice != nil && status.Convert(errDevice).Code() == codes.NotFound) ||
+			(getProtocolState(deviceTopo) == topodevice.ServiceState_UNAVAILABLE) {
```

Comment:
this seems very closely related to line 160 below - is there an overlap?

Reply:
thanks for spotting